### PR TITLE
fix parsing arrow function using php_version dynamically into parser

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**
In this PR I removed the PHP version fixed for the emulator, and now we are using the PHP version dynamically.
Removing PHP version specification break all the tests.

Related issue: #160

### CODE REVIEW

File: src/Analyzer/FileParser.php [hunk 1]
```diff
@@ -26,7 +25,7 @@ public function __construct(NodeTraverser $traverser, FileVisitor $fileVisitor,
 
         $lexer = new Emulative([
             'usedAttributes' => ['comments', 'startLine', 'endLine', 'startTokenPos', 'endTokenPos'],
-             'phpVersion' => self::PHP_VERSION,
+             'phpVersion' => phpversion('tidy'),
```
Comment (micheleorselli) [inline src/Analyzer/FileParser.php:28]: tidy is not needed

**Comment by micheleorselli** [comment]
looks good to me though I was wondering if this could cause problems 🤔

File: src/Analyzer/FileParser.php (hunk 1 above)
Comment (AlessandroMinoccheri) [reply src/Analyzer/FileParser.php:28]: for the coding standard, we need to use it.

**Comment by AlessandroMinoccheri** [comment]
The problem @micheleorselli could be with the generated `.phar`? At the moment we use 7.1 always, now with this PR the PHP version depends on the PHP installed.
We can add multiple `.phar` files with different PHP versions.
At the moment I can't figure out other problems but we can try arkitect on different projects.

**Comment by micheleorselli** [comment]
Hi,

> The problem @micheleorselli could be with the generated `.phar`? At the moment we use 7.1 always, now with this PR the PHP version depends on the PHP installed.

yeah I think so

> We can add multiple `.phar` files with different PHP versions.

Not really sure we want to do that, we should stick with the minimum php supported version. Still, the phar should be able to work with newver php versions

> At the moment I can't figure out other problems but we can try arkitect on different projects.

File: src/Analyzer/FileParser.php (hunk 1 above)
Comment (AlessandroMinoccheri) [reply src/Analyzer/FileParser.php]: You are right, I have removed the extension @micheleorselli

**Comment by micheleorselli** [comment]
We can merge and release this. I guess the same issue potentially applies to php 8 and 8.1. In the meantime we can merge this

**Pull request merged**